dev/release/download_rc_binaries.py

```diff
@@ -121,17 +121,28 @@ def _download_url(self, url, dest_path, *, extra_args=None):
             dest_path,
             url,
         ]
-        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        stdout, stderr = proc.communicate()
-        if proc.returncode != 0:
-            try:
-                # Don't leave possibly partial file around
-                os.remove(dest_path)
-            except IOError:
-                pass
-            raise Exception(f"Downloading {url} failed\n"
-                            f"stdout: {stdout}\nstderr: {stderr}")
+        # Retry subprocess in case it fails with OpenSSL Connection errors
+        # https://issues.apache.org/jira/browse/INFRA-25274
+        for attempt in range(5):
+            if delay := attempt * 3:
```

Comment:
The following may be more readable :-)

```suggestion
            if attempt > 0:
                delay = attempt * 3:
```

Reply:
But either of them is OK to me.